### PR TITLE
Integrate clone-job into bootstrap

### PR DESCRIPTION
**Pull request by b10n1k**
- Add clone-job at the end of the bootstrap script, in case there are parameters passed in this. the script will expands them after `clone_job.pl -from·openqa.opensuse.org`.

### CODE REVIEW

File: script/openqa-bootstrap [hunk 1]
```diff
@@ -5,10 +5,15 @@ dbname="${dbname:="openqa"}"
...
+    return 1
+}
+
 source /etc/os-release
-if [[ "$NAME" == "openSUSE Leap" ]] ; then
-	zypper -n addrepo -p 90 obs://devel:openQA devel:openQA
-	zypper -n addrepo -p 91 obs://devel:openQA:Leap:${VERSION} devel:openQA:Leap:${VERSION}
+if [[ "$NAME" == "openSUSE Leap"]] ; then
+        if ! foundrepo devel:openQA; then zypper -n addrepo -p 90 obs://devel:openQA devel:openQA; fi
+        if ! foundrepo devel:openQA:Leap:${VERSION}; then zypper -n addrepo -p 91 obs://devel:openQA:Leap:${VERSION} devel:openQA:Leap:${VERSION}; fi
```
Comment (okurz) [inline script/openqa-bootstrap]: While I think this is a nice idea please keep the above change for a separate PR. The challenge is that we want to keep `script/openqa-bootstrap` as easy as possible to read for humans to act as "instructions". Also you are now mixing tabs and spaces. Better to keep the existing style. I have a change prepared to use a consistent spaces-only file as we also have for other files to fix this though.

File: script/openqa-bootstrap [hunk 2]
```diff
@@ -105,3 +110,10 @@ EOF
 
 # start worker
 systemctl enable --now openqa-worker@1.service
+
+# clone jobs if jobId(s) are given
+if [ $# -ne 0 ]; then
+    for arg; do
+	sudo /var/lib/openqa/script/clone_job.pl -from openqa.opensuse.org "$arg"
```
Comment (okurz) [inline script/openqa-bootstrap]: This should be at best just `openqa-clone-job "$@"`. I would not call clone jobs multiple times here, *however* it is a neat idea. We could extend this actually in `openqa-clone-job` itself to iterate over each job-url or id.

I would simply forward *all* additional arguments so that we can also pass further job parameters to overwrite the settings on the cloned job. Example call:

```
openqa-bootstrap --from https://openqa.opensuse.org 12345 SCHEDULE=tests/boot/boot_to_desktop,tests/x11/kontact
```

Also, please no "sudo". There should be no root-permissions required. The authentication is handled by the api key+secret, not unix permissions.

File: script/openqa-bootstrap [hunk 3]
```diff
@@ -9,7 +9,7 @@ source /etc/os-release
 if [[ "$NAME" == "openSUSE Leap" ]] ; then
 	zypper -n addrepo -p 90 obs://devel:openQA devel:openQA
 	zypper -n addrepo -p 91 obs://devel:openQA:Leap:${VERSION} devel:openQA:Leap:${VERSION}
-	zypper -n  --gpg-auto-import-keys refresh
+	zypper -n --gpg-auto-import-keys refresh
```
Comment (okurz) [inline script/openqa-bootstrap]: unrelated change

File: script/openqa-bootstrap [hunk 4]
```diff
@@ -102,6 +102,18 @@ key = ${API_KEY}
 secret = ${API_SECRET}
 EOF
 
+# configure worker@1 to use x86_64 backends
+cat >> /etc/openqa/workers.ini <<EOF
+[1]
+#WORKER_CLASS = qemu_x86_64_staging,qemu_x86_64
+EOF
+
```
Comment (okurz) [inline script/openqa-bootstrap]: again, unfortunately unrelated change. And also the line is commented. I guess you just did that as an experiment to try out?

File: script/openqa-bootstrap [hunk 5]
```diff
@@ -102,6 +102,18 @@ key = ${API_KEY}
...
 
 # start worker
 systemctl enable --now openqa-worker@1.service
+
+# clone jobs if jobId is given. Possible to pass extra arguments as well.
+# ex: openqa-bootstrap 12345 SCHEDULE=tests/boot/boot_to_desktop,tests/x11/kontact
+if [ $# -ne 0 ]; then
+	/var/lib/openqa/script/clone_job.pl -from openqa.opensuse.org "$@"
```
Comment (okurz) [inline script/openqa-bootstrap]: Please use `openqa-clone-job` and delete the `-from openqa.opensuse.org` which would hardcode o3. When we pass all parameters it is not a problem to pass the full `--from https://openqa.opensuse.org <job_id>` as well

Comment (b10n1k) [reply script/openqa-bootstrap]: updated

File: script/openqa-bootstrap (hunk 4 above)
Comment (b10n1k) [reply script/openqa-bootstrap]: i removed the # in the beginning. i dont think this is unrelated. The job never runs if this is not set

File: script/openqa-bootstrap (hunk 3 above)
Comment (b10n1k) [reply script/openqa-bootstrap]: yes it is but also trivial.i dont think it 's worth separate ticket. i have restored it for now

Comment (okurz) [reply script/openqa-bootstrap]: Sure, I only meant a separate *commit*, you can still have it in this PR of course

File: script/openqa-bootstrap (hunk 4 above)
Comment (okurz) [reply script/openqa-bootstrap]: qemu_x86_64 should be the default. We have Tumbleweed tests to cover this. Can you show which job fails?

File: script/openqa-bootstrap [hunk 6]
```diff
@@ -102,6 +102,18 @@ key = ${API_KEY}
...
+EOF
+
 
 # start worker
 systemctl enable --now openqa-worker@1.service
+
+# clone jobs if jobId is given. Possible to pass extra arguments as well.
+# ex: openqa-bootstrap -from openqa.opensuse.org 12345 SCHEDULE=tests/boot/boot_to_desktop,tests/x11/kontact
+if [ $# -ne 0 ]; then
+	/var/lib/openqa/script/openqa-clone-job "$@"
```
Comment (okurz) [inline script/openqa-bootstrap]: Just the application from path, w/o any directory prefix, please

File: script/openqa-bootstrap (hunk 4 above)
Comment (b10n1k) [reply script/openqa-bootstrap]: No really, because i work on a virtual machine to avoid to touch my openqa. the workers.ini have commented out the workers and the jobs are scheduled but never run if i dont set up the file as such. i assume that you mean that the job picks x86_64 even when the workers are commented somehow.

File: script/openqa-bootstrap (hunk 6 above)
Comment (b10n1k) [reply script/openqa-bootstrap]: done

File: script/openqa-bootstrap (hunk 4 above)
Comment (okurz) [reply script/openqa-bootstrap]: yes

**Comment by okurz** [comment]
Thank you. Congrats, hackweek project completed! ;)
